### PR TITLE
237 POST /email/sendUserViolation fix status code if invalid email

### PR DESCRIPTION
**Pull request by Rafaelkaaa**
https://github.com/OlhenShu/GreenCityProjectStage/issues/237

### CODE REVIEW

**Review by urio999 — APPROVED** [review]
Well Done

File: core/src/main/java/greencity/controller/EmailController.java [hunk 1]
```diff
@@ -111,6 +111,7 @@ public ResponseEntity<Object> sendHabitNotification(@RequestBody SendHabitNotifi
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = HttpStatuses.OK),
             @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+            @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND),
```
Comment (DenysLiubchenko) [inline core/src/main/java/greencity/controller/EmailController.java]: You need to delete coma here

File: core/src/main/java/greencity/config/SecurityConfig.java [hunk 2]
```diff
@@ -139,6 +139,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 "/user/shopping-list-items",
                 "/user/{userId}/habit",
                 "/ownSecurity/set-password",
+                "/email/sendUserViolation")
```
Comment (DenysLiubchenko) [inline core/src/main/java/greencity/config/SecurityConfig.java]: Here you need to alter ')' into ','

**Review by DenysLiubchenko — CHANGES_REQUESTED** [review]

File: service/src/main/java/greencity/service/EmailServiceImpl.java [hunk 3]
```diff
@@ -280,6 +280,9 @@ public void sendMessageOfActivation(UserActivationDto userActivationDto) {
 
     @Override
     public void sendUserViolationEmail(UserViolationMailDto dto) {
+        if(!dto.getEmail().matches("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$")){
```
Comment (lagysha) [inline service/src/main/java/greencity/service/EmailServiceImpl.java:283]: You can use on dto field @Email annotation instead of validating here

**Pull request merged**